adi/adt7420.py

```diff
@@ -19,10 +19,28 @@ def __init__(self, uri=""):
         if not self._ctrl:
             raise Exception("ADT7420 device not found")
 
-        self.temp = self._channel(self._ctrl, "temp")
+        self.channel = []
+        self._rx_channel_names = []
 
-    class _channel(attribute):
-        """ADT7420 Channel"""
+        for ch in self._ctrl.channels:
+            name = ch._id
+
+            self._rx_channel_names.append(name)
+            self.channel.append(name)
+
+            if name == "temp":
+                # no-OS
+                setattr(self, name, self.channel_temp(self._ctrl, name))
+                self.temp = self.channel_temp(self._ctrl, "temp")
+            elif name == "temp1":
+                # Linux
+                setattr(self, name, self.channel_temp1(self._ctrl, name))
+                self.temp = self.channel_temp1(self._ctrl, "temp1")
+            else:
+                raise Exception(f"Unsupported: {name}")
```

Comment:
alittle confused on this loop. Do all the channels have the same name? Is there more than one channel on the device? Maybe an emulated context will help